includes/forms/class-wp-job-manager-form-submit-job.php

```diff
@@ -1209,4 +1213,94 @@ public function track_job_submission( $post_id ) {
+	 * @return bool
+	 */
+	public function is_renew_action() {
+		$job_id = isset( $_GET['job_id'] ) ? sanitize_text_field( wp_unslash( $_GET['job_id'] ) ) : '';
+		$nonce  = isset( $_GET['nonce'] ) ? wp_unslash( $_GET['nonce'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Nonce should not be modified.
+		$action = 'job_manager_renew_job_' . $job_id;
+		if ( ! wp_verify_nonce( $nonce, $action ) ) {
+			return false;
+		}
+		return isset( $_GET['action'] ) && sanitize_text_field( wp_unslash( $_GET['action'] ) ) === 'renew';
+	}
+
+	/**
+	 * Handle steps for renewing a listing before expiry, removes edit steps.
+	 *
+	 * @access private
+	 * @param array $steps
+	 * @return array
+	 */
+	public function remove_edit_steps_for_renewal( $steps ) {
+		unset( $steps['submit'], $steps['preview'] );
+		$steps['renew-listing'] = [
+			'name'     => 'Renew Listing',
+			'view'     => [ $this, 'preview' ],
+			'handler'  => [ $this, 'renew_preview_handler' ],
+			'priority' => 29, // This action should be just before 'done' whose priority is 30.
+		];
+		return apply_filters( 'renew_job_steps', $steps, $this );
+	}
+
+	/**
+	 * Handles the renew-listing form submission.
+	 *
+	 * @throws Exception On validation error.
+	 */
+	public function renew_preview_handler() {
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Input is used safely.
+		if ( empty( $_POST ) ) {
+			return;
+		}
+
+		$this->check_preview_form_nonce_field();
+
+		if ( ! empty( $_POST['continue'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing -- Input is used safely.
+			$job      = get_post( $this->job_id );
+			$renewing = $this->is_renew_action() && job_manager_job_can_be_renewed( $this->job_id );
+
+			if ( $renewing ) {
+				$old_expiry = date_create_immutable_from_format( 'Y-m-d', get_post_meta( $job->ID, '_job_expires', true ) );
+				$new_expiry = calculate_job_expiry( $job->ID, false, $old_expiry );
+				update_post_meta( $job->ID, '_job_expires', $new_expiry );
+
+				/**
+				 * Fires when a job listing status is about to be updated.
+				 *
+				 * @param int  $job_id The job ID.
+				 * @param bool $renewing Whether the job is being renewed.
+				 */
+				$post_status = apply_filters( 'submit_job_post_status', 'publish', $job, true );
+
+				$update_job                  = [];
+				$update_job['ID']            = $job->ID;
+				$update_job['post_status']   = $post_status;
+				$update_job['post_date']     = current_time( 'mysql' );
+				$update_job['post_date_gmt'] = current_time( 'mysql', 1 );
+
+				wp_update_post( $update_job );
+			}
+
+			$this->step ++;
+		}
+	}
+
+	/**
+	 * Filters submit button text for renew-listing step.
+	 *
+	 * @param int $text The button text.
+	 * @return string
+	 */
+	public function submit_button_text_renewal( $text ) {
+		if ( $this->get_step_key( $this->step ) === 'renew-listing' ) {
+			return __( 'Renew Listing &rarr;', 'wp-job-manager' );
```

Comment:
Was the arrow intentional? Not sure if it is a good idea

Reply:
Yeah it was intentional, why do you think it's not a good idea?

Reply:
It feels a bit out of place and it isn't consistent with the other submit button. Seems more like a customization that some sites might want to do by updating the filter. But let's check with @yscik or @Luchadores 

So we are looking if we want to add an arrow in the submit button which will be the last step of renewing a listing. You can have a look at the video in the description to see the screen that is after that step. Do you think that we should add it?

![Screenshot 2023-05-23 at 11 13 45](https://github.com/Automattic/WP-Job-Manager/assets/53191348/b27cbf15-5534-40bb-97f6-cfd278380bd3)

For reference, this is the same step when we relist a previously expired listing:

![Screenshot 2023-05-23 at 11 17 11](https://github.com/Automattic/WP-Job-Manager/assets/53191348/06b86a0d-5f2c-45b0-8706-093eb44b2fde)

Reply:
We had a discussion about this one [here](https://a8c.slack.com/archives/C036R3Y5ET0/p1684832723675199) and we are going to keep the button.